### PR TITLE
Nv 2231 Template Store - template store analytics

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

We need to be able to answer the following questions with the new events

Key Event: How many people create a workflow from a template store
[1] event on template creation from the template store.

What are the most popular templates in the store?
[1] event on template creation from the template store.
[2] event on the template click on the template store.

How many people opened the template store but haven’t used it?
[3] event on the template store opened.
[1] event on template creation from the template store.

Did the average number of templates for an account go higher after the release of the feature
[4] event on template creation (we probably already have this one)

NOTE: I added a couple of events we could add in order to answer those questions. I numbered them so we can see that there are the same events that we need for different questions.

[2] === [3]

### CODE REVIEW

File: apps/web/src/pages/templates/TemplatesListNoData.tsx [hunk 1]
```diff
@@ -154,7 +168,14 @@ export const TemplatesListNoData = ({
...
                     data-test-id="second-workflow-tile"
                     disabled={readonly || isCreating}
-                    onClick={() => onTemplateClick(template)}
+                    onClick={() => {
+                      segment.track('[Template Store] Click Create Notification Template', {
+                        templateIdentifier: template?.triggers[0]?.identifier || '',
+                        location: TemplateCreationSourceEnum.EMPTY_STATE,
+                      });
+
+                      onTemplateClick(template);
```
Comment (LetItRock) [inline apps/web/src/pages/templates/TemplatesListNoData.tsx:177]: we will have 2 analytic events: one from here and one triggered on the BE, is that what we want?

File: apps/web/src/api/hooks/notification-templates/useCreateTemplateFromBlueprint.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ const mapBlueprintToTemplate = (blueprint: INotificationTemplate): ICreateNotifi
   critical: blueprint.critical,
   preferenceSettings: blueprint.preferenceSettings,
   blueprintId: blueprint._id,
+  __source: blueprint.__source,
```
Comment (LetItRock) [inline apps/web/src/api/hooks/notification-templates/useCreateTemplateFromBlueprint.ts:17]: The `__source` is not a part of the `blueprint/template` it's something additional - metadata, then I feel like we shouldn't send it in the body, but in the query params instead.

File: apps/web/src/components/quick-start/in-app-onboarding/TriggerNode.tsx [hunk 3]
```diff
@@ -80,6 +81,7 @@ function TriggerButton({ setOpened }: { setOpened: (value: boolean) => void }) {
     async function createOnBoardingTemplate() {
       const payloadToCreate = {
         notificationGroupId: groups[0]._id,
+        isBlueprint: false,
```
Comment (LetItRock) [inline apps/web/src/components/quick-start/in-app-onboarding/TriggerNode.tsx:84]: do we have to set it?

Comment (djabarovgeorge) [reply apps/web/src/components/quick-start/in-app-onboarding/TriggerNode.tsx:84]: Added this flag to make sure that there won't be a case where this template creates by mistake which will result in a blueprint in the Template Store

File: apps/web/src/pages/templates/TemplatesListNoData.tsx (hunk 1 above)
Comment (djabarovgeorge) [reply apps/web/src/pages/templates/TemplatesListNoData.tsx:177]: That is true @LetItRock , was not sure if we want to separate those api-clients events or not. do you think we can drop this client event?

File: apps/web/src/api/hooks/notification-templates/useCreateTemplateFromBlueprint.ts (hunk 2 above)
Comment (djabarovgeorge) [reply apps/web/src/api/hooks/notification-templates/useCreateTemplateFromBlueprint.ts]: Done :)

File: apps/web/src/pages/templates/TemplatesListNoData.tsx (hunk 1 above)
Comment (djabarovgeorge) [reply apps/web/src/pages/templates/TemplatesListNoData.tsx:177]: Will merge for now we always can remove it.

Comment (djabarovgeorge) [reply apps/web/src/pages/templates/TemplatesListNoData.tsx:177]: Let's merge this for now, we can always remove it.

Comment (djabarovgeorge) [reply apps/web/src/pages/templates/TemplatesListNoData.tsx:177]: Let's merge this for now, we can always remove it.
@LetItRock  @scopsy

**Review by LetItRock — APPROVED** [review]

**Pull request merged**